lib/active_merchant/billing/gateways/mercado_pago.rb

```diff
@@ -196,6 +198,46 @@ def add_payment(post, options)
         post[:payment_method_id] = options[:payment_method_id] if options[:payment_method_id]
       end
 
+      def add_net_amount(post, options)
+        post[:net_amount] = options[:net_amount].to_f if options[:net_amount]
+      end
+
+      def add_taxes(post, options)
+        return unless (tax_object = options[:taxes])
+
+        if tax_object.is_a?(Array)
+          post[:taxes] = process_taxes_array(tax_object)
+        elsif tax_object.is_a?(Hash)
+          post[:taxes] = process_taxes_hash(tax_object)
+        else
+          raise taxes_error
+        end
+      end
+
+      def process_taxes_hash(tax_object)
+        [sanitize_taxes_hash(tax_object)]
+      end
+
+      def process_taxes_array(taxes_array)
+        taxes_array.map do |tax_object|
+          raise taxes_error unless tax_object.is_a?(Hash)
+          sanitize_taxes_hash(tax_object)
+        end
+      end
+
+      def sanitize_taxes_hash(tax_object)
+        tax_value = tax_object['value'] || tax_object[:value]
+        tax_type = tax_object['type'] || tax_object[:type]
```

Comment:
Did you notice in your tests that these were inconsistent in whether they were symbols or strings?

Reply:
Since these fields are nested inside of the hash, they will come in as strings from an upstream API. For example,  the Adyen `split` field takes these only as strings:

https://github.com/activemerchant/active_merchant/blob/master/test/unit/gateways/adyen_test.rb#L342-L366

This is a bit confusing since your test payload alternates between strings and symbols. I thought that accepting either could save some grief in the future but am open to suggestions here.

Reply:
I was mostly wondering if you saw something in particular. The strings vs. symbols was very confusing for me in that code, and this seems like a great way to ensure that the correct values are coming through!